tests/test_io.py

```diff
@@ -0,0 +1,217 @@
+def create_test_env(tmp_path_factory):
+    # Create an ideal observation file using tmp_path_factory
+    pthList = ["Spectra", "Resistance", "S11"]
+    s11List = [
+        "Ambient",
+        "AntSim3",
+        "HotLoad",
+        "LongCableOpen",
+        "LongCableShorted",
+        "ReceiverReading01",
+        "ReceiverReading02",
+        "SwitchingState01",
+        "SwitchingState02",
+    ]
+    a = tmp_path_factory.mktemp("Test_Obs")
+    b = a / "Receiver01_2020_01_01_010_to_200MHz"
+    b.mkdir()
+    d = b / "25C"
+    d.mkdir()
+    note = d / "Notes.txt"
+    note.touch()
+    dlist = []
+    slist = []
+    for i, p in enumerate(pthList):
+        dlist.append(d / p)
+        dlist[i].mkdir()
+        if p == "Spectra":
+            print("Making Spectra files")
+            fileList = [
+                "Ambient",
+                "AntSim3",
+                "HotLoad",
+                "LongCableOpen",
+                "LongCableShorted",
+            ]
+            for filename in fileList:
+                # print(filename)
+                name1 = filename + "_01_2020_001_01_01_01_lab.acq"
+                file1 = dlist[i] / name1
+                # print(file1)
+                file1.touch()
+        if p == "Resistance":
+            print("Making Resistance files")
+            fileList = [
+                "Ambient",
+                "AntSim3",
+                "HotLoad",
+                "LongCableOpen",
+                "LongCableShorted",
+            ]
+            for filename in fileList:
+                # print(filename)
+                name1 = filename + "_01_2020_001_01_01_01_lab.csv"
+                file1 = dlist[i] / name1
+                # print(file1)
+                file1.touch()
+        if p == "S11":
+            print("Making S11 files")
+            fileList = ["External", "Match", "Open", "Short"]
+            for k, s in enumerate(s11List):
+                slist.append(dlist[i] / s)
+                slist[k].mkdir()
+                if s[:-2] == "ReceiverReading":
+                    fileList = ["ReceiverReading", "Match", "Open", "Short"]
+                elif s[:-2] == "SwitchingState":
+                    fileList = [
+                        "ExternalOpen",
+                        "ExternalMatch",
+                        "ExternalShort",
+                        "Match",
+                        "Open",
+                        "Short",
+                    ]
+                else:
+                    fileList = ["External", "Match", "Open", "Short"]
+                for filename in fileList:
+                    # print(filename)
+                    name1 = filename + "01.s1p"
+                    name2 = filename + "02.s1p"
+                    file1 = slist[k] / name1
+                    file1.write_text(
+                        "# Hz S RI R 50\n40000000        0.239144887761343       0.934085904901478\n40000000        0.239144887761343       0.934085904901478"
```

Comment:
To make it easier to read, you could use triple quotes here. Another idea would be to have the file already existing in the tests directory (under `tests/data/`) and just copy it into the temp directory you're making here.